apps/elixir_chatbot_core/lib/index_server.ex

```diff
@@ -85,8 +105,15 @@ defmodule ElixirChatbotCore.IndexServer do
     {:reply, res, index}
   end
 
-  defp create_index_path(docs_db, model_name) do
+  @impl true
+  def handle_call({:lookup, text, k}, _from, index) do
+    res = SimilarityIndex.lookup(index, text, k: k)
+    {:reply, res, index}
+  end
+
+  defp create_index_path(docs_db, model_name, metric) do
     model_name = String.replace(model_name, "/", "-")
-    "#{Application.fetch_env!(:chatbot, :hnsw_index_path)}-#{docs_db}_#{model_name}"
+
+    "#{Application.fetch_env!(:chatbot, :hnsw_index_path)}-#{docs_db}_#{model_name}_#{Atom.to_string(metric)}"
```

Comment:
All our embedding models use normalized vectors so the ranking will be the same for L2 and cosine metrics. I think we can abandon testing metrics at all and use cosine all the time.

Reply:
Hm, some of the models I wanted to try seemed to prefer cosine over l2, but i think in the end they werent supported by Bumblebee. I think its good to leave the functionality just in case, but in that case i will only test l2 from now on - 50% less indexing will def make things faster

Reply:
As for why the model name takes cosine into account, afaik hnsw creates different skipgraphs for different metrics, so its important to differentiate the cache file

Reply:
Ok, I didn't know that index needs to know about it. So leave the whole functionality but test only one metrics.

But test rather cosine than l2. OpenAI recommends it (here: https://platform.openai.com/docs/guides/embeddings/limitations-risks) so we'll be able to justify the choice easily